IanGRap/Squish
# Player Selection

Contributors: Alfredo Cuevas, Austin Kellar

Goal: Create a system for selecting the number of players and starting a match with the correct number of players, all hooked up to the correct controller

Important Files Changed:
- NEW SCENE: Player Selection/Player Selection Screen (contains player select UI and logic for mapping players to controllers)
- NEW SCENE: Player Selection/player selection game test (a copy of sprint 2 demo which has four players in it)
- SceneController.cs (a static object which allows you to choose which scene gets started when players 
press Start in the player selection screen)
- ActivePlayers.cs (a static class that can be accessed globally which contains information about which players are signed in to play in the next match)
- UIController.cs (Controls the UI text in the player selection scene)
- ControllerManager.cs (Assigns controllers to players)
- Controller.cs (Holds information about controllers)
- Player.cs (Holds information about players in relation to what controller they are assigned)
- PlayerManager.cs (Adds and removes players from ActivePlayers.cs and updates the player select UI)
- PlayerInput.cs (On awake, if there is no player matching the player's assigned PlayerNumber inside ActivePlayers.cs, delete that player)
- ControllerManager.cs (Make sure that the camera is not trying to track null objects)

How to Test:
- Load the scene Player Selection Screen
- Connect 2-4 players and press Start
- Ensure that the player selection game test scene is loaded with the correct number of players
- Ensure that for each controller that joined the match, that controller move exactly 1 of the players in the match

Notes:
- You have to choose what scene will start when you press Start (inside the Scene Controller object)
    - The scene that you choose needs to exist in the Build Settings or you will get a console error
- We got rid of the input fields inside PlayerInput.cs and replaced them with a single field called PlayerNumber. All previously existing scenes that have Players in them WILL NOW BE BROKEN. You will have to assign each player in the scene a player number from 1 to 4 in order for controller input to be recognized.
- There is still the bug where if you disconnect controllers and reconnect them, unity will think that they are a different controller besides 1, 2, 3, 4. We have a potential fix in mind for this but it will take a lot of time to implement so we think it should be its own PR
